app/controllers/users_controller.rb

```diff
@@ -1,9 +1,11 @@
 class UsersController < ApplicationController
   def index
-    @users = User.all
+    # Eager loading posts for all users
+    @users = User.all.includes(:posts)
   end
 
   def show
-    @user = User.find(params[:id])
+    # Eager loading posts, comments, and authors for the specific user
+    @user = User.includes(posts: [{ comments: :author }]).find(params[:id])
   end
```

Comment:
- [ ] I was able to see the home page (users page) without any login/signup. Users should not type the URL for signup/login to access them. Kindly add a line of code (from devise) that prevents users from seeing pages of the app before login/signup.